numpy/core/einsumfunc.py

```diff
@@ -1101,13 +1104,19 @@ def einsum(*operands, **kwargs):
         if specified_out and ((num + 1) == len(contraction_list)):
             handle_out = True
 
-        # Call tensordot
+        # Handle broadcasting vs BLAS cases
         if blas:
-
             # Checks have already been handled
             input_str, results_index = einsum_str.split('->')
             input_left, input_right = input_str.split(',')
+            if 1 in tmp_operands[0] or 1 in tmp_operands[1]:
+                if any(tmp_operands[0].shape[input_left.index(ind)] !=
+                       tmp_operands[1].shape[input_right.index(ind)]
+                       for ind in idx_rm):
```

Comment:
To reiterate my now-hidden comment, I think what you had before was better - sorry!

Reply:
That's fine, easy enough to remove the commit